axelor-project/src/main/java/com/axelor/apps/project/service/ProjectTaskToolServiceImpl.java

```diff
@@ -32,4 +63,107 @@ public Optional<TaskStatus> getCompletedTaskStatus(
     }
     return completedTaskStatus;
   }
+
+  @Override
+  public List<ProjectTask> getProjectTaskDependingStatus(
+      TaskStatus taskStatus, Integer taskStatusManagementSelect, Long objectId) {
+    StringBuilder filter = new StringBuilder();
+    Map<String, Object> bindings = new HashMap<>();
+
+    filter.append("self.status = :taskStatus ");
+    bindings.put("taskStatus", taskStatus);
+
+    if (taskStatusManagementSelect != null) {
+      if (ProjectRepository.TASK_STATUS_MANAGEMENT_APP == taskStatusManagementSelect) {
+        filter.append(
+            "AND (self.project.taskStatusManagementSelect = :taskStatusManagementSelect "
+                + "OR (self.project.taskStatusManagementSelect = :categoryTaskStatusManagementSelect AND self.projectTaskCategory IS NULL))");
+        bindings.put("taskStatusManagementSelect", taskStatusManagementSelect);
+        bindings.put(
+            "categoryTaskStatusManagementSelect",
+            ProjectRepository.TASK_STATUS_MANAGEMENT_CATEGORY);
+      } else {
+        filter.append("AND self.project.taskStatusManagementSelect = :taskStatusManagementSelect ");
+        bindings.put("taskStatusManagementSelect", taskStatusManagementSelect);
+
+        if (ProjectRepository.TASK_STATUS_MANAGEMENT_PROJECT == taskStatusManagementSelect) {
+          filter.append("AND self.project.id = :id ");
+          bindings.put("id", objectId);
+        } else if (ProjectRepository.TASK_STATUS_MANAGEMENT_CATEGORY
+            == taskStatusManagementSelect) {
+          filter.append("AND self.projectTaskCategory.id = :id ");
+          bindings.put("id", objectId);
+        }
+      }
+    }
+
+    return projectTaskRepository.all().filter(filter.toString()).bind(bindings).order("id").fetch();
+  }
+
+  @Override
+  public List<ProjectTask> getProjectTaskToUpdate(AppProject appProject) {
+    if (appProject == null || appProject.getId() == null) {
+      return new ArrayList<>();
+    }
+    AppProject savedAppProject = appProjectRepository.find(appProject.getId());
+    return getProjectTaskListToUpdate(
+        savedAppProject.getDefaultTaskStatusSet(),
+        appProject.getDefaultTaskStatusSet(),
+        ProjectRepository.TASK_STATUS_MANAGEMENT_APP,
+        null);
+  }
+
+  @Override
+  public List<ProjectTask> getProjectTaskListToUpdate(
```

Comment:
This method should not be public and in the interface if it is not called from other services.